components/win-users/src/account.rs

```diff
@@ -38,6 +40,85 @@ impl Account {
     pub fn from_name_and_system(name: &str, system_name: &str) -> Option<Account> {
         lookup_account(name, Some(system_name.to_string()))
     }
+
+    pub fn from_sid(sid: &str) -> Option<Account> {
+        let sid = if let Ok(sid) = Sid::from_str(sid) {
+            sid
+        } else {
+            return None;
+        };
+        let mut name_size: u32 = 0;
+        let mut domain_size: u32 = 0;
+        unsafe {
+            LookupAccountSidW(null(),
+                              sid.raw.as_ptr() as *mut _,
+                              null_mut(),
+                              &mut name_size as LPDWORD,
+                              null_mut(),
+                              &mut domain_size as LPDWORD,
+                              null_mut())
+        };
+        match Error::last_os_error().raw_os_error().unwrap() as u32 {
```

Comment:
Not all windows functions are guaranteed to clear last_os_error on success. Maybe the rust last_os_error enforces this somehow, but better pattern is to check return value and then call last_os_error if failure is indicated.

Reply:
Thanks for catching this! Fixed in [5a0bcd](https://github.com/habitat-sh/habitat/pull/7224/commits/5a0bcd344ab34636b2a44d59d70ab162cd7f0fed)